slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -28,15 +28,87 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
                         "GetMap is not yet implemented.");
 }
 
-void SLAMServiceImpl::CreateMap() {
-    mapping::MapBuilder mapBuilder;
+SLAMServiceActionMode SLAMServiceImpl::GetActionMode() {
```

Comment:
"Action mode" is something I came up with to distinguish the normal mode ("rgbd", "mono", "2d") to the "modes" that define whether or not we run in localization, mapping, or updating "mode". The name can be changed if someone has a different idea/preference.